src/coreclr/System.Private.CoreLib/src/System/Variant.cs

```diff
@@ -78,323 +23,311 @@ internal partial struct Variant
         [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "Variant_ConvertOleColorToSystemColor")]
         internal static partial void ConvertOleColorToSystemColor(ObjectHandleOnStack objret, uint value, IntPtr pMT);
 
-        //
-        // Native Methods
-        //
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        internal extern void SetFieldsObject(object val);
-
-        //
-        // Constructors
-        //
-
-        internal Variant(int flags, object or, long data)
-        {
-            _flags = flags;
-            _objref = or;
-            _data = data;
-        }
-
-        public Variant(bool val)
-        {
-            _objref = null;
-            _flags = CV_BOOLEAN;
-            _data = (val) ? bool.True : bool.False;
-        }
-
-        public Variant(sbyte val)
-        {
-            _objref = null;
-            _flags = CV_I1;
-            _data = val;
-        }
-
-        public Variant(byte val)
-        {
-            _objref = null;
-            _flags = CV_U1;
-            _data = val;
-        }
-
-        public Variant(short val)
-        {
-            _objref = null;
-            _flags = CV_I2;
-            _data = val;
-        }
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "Variant_ConvertValueTypeToRecord")]
+        private static partial void ConvertValueTypeToRecord(ObjectHandleOnStack obj, out ComVariant pOle);
 
-        public Variant(ushort val)
+        internal static ComVariant GetComIPFromObjectRef(object? obj)
         {
-            _objref = null;
-            _flags = CV_U2;
-            _data = val;
+            IntPtr pUnk = GetIUnknownOrIDispatchForObject(ObjectHandleOnStack.Create(ref obj), out bool isIDispatch);
```

Comment:
I would assert this is `Marshal.IsComObject()` or `ComWrappers.TryGetComInstance()`.

Reply:
Oh, nevermind. This is doing something different. This name is a legacy one. I would instead do something along the lines of `GetIUnknownOrIDispatchForObject`. The entire notion of "ComIP" is confusing in the runtime.